File: model/product.model.js
const db = require('../config/db');
const mongoose = require('mongoose');
const UserModel = require('../model/user.model');
const { Schema } = mongoose;

// Định nghĩa schema cho sản phẩm
const productSchema = new Schema({
    userId: {
        type: Schema.Types.ObjectId,
        ref: UserModel.modelName, // Liên kết với model User
        required: true,
    },
    name: {
        type: String,
        required: true,
    },
    description: {
        type: String,
        required: true,
    },
    price: {
        type: Number,
        required: true,
    },
    quantity: {
        type: Number,
        required: true,
    },
}, { timestamps: true }); // Thêm timestamps để theo dõi thời gian tạo và cập nhật

// Tạo model từ schema
const ProductModel = db.model('Product', productSchema);

module.exports = ProductModel;

File: managestore/lib/models/product_model.dart
class Product {
  final String id;
  final String name;
  final String description;
  final double price;
  final int quantity;

  Product({
    required this.id,
    required this.name,
    required this.description,
    required this.price,
    required this.quantity,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['_id'],
      name: json['name'],
      description: json['description'],
      price: json['price'].toDouble(),
      quantity: json['quantity'],
    );
  }
}
